gke/cluster/main.tf

```diff
@@ -1,251 +1,43 @@
-      dynamic "master_global_access_config" {
-        for_each = try([private_cluster_config.value.masterGlobalAccessConfig], [])
-        content {
-          enabled = try(private_cluster_config.value.masterGlobalAccessConfig.enabled, false)
-        }
-      }
-    }
-  }
-
-  dynamic "release_channel" {
-    for_each = try([var.release_channel], [])
-
-    content {
-      channel = release_channel.value
-    }
-  }
-
-  min_master_version = var.kubernetes_version
-
-  addons_config {
-    http_load_balancing {
-      disabled = try(var.addons_config.httpLoadBalancing.disabled, false)
-    }
-    horizontal_pod_autoscaling {
-      disabled = try(var.addons_config.horizontalPodAutoscaling.disabled, false)
-    }
-  }
-
-  vertical_pod_autoscaling {
-    enabled = var.enable_vertical_pod_autoscaling
-  }
-
-  cluster_autoscaling {
-    auto_provisioning_defaults {
-      service_account   = try(var.cluster_autoscaling.autoprovisioningNodePoolDefaults.serviceAccount, null)
-      boot_disk_kms_key = try(var.cluster_autoscaling.autoprovisioningNodePoolDefaults.bootDiskKmsKey, null)
-    }
-  }
-
-  dynamic "master_authorized_networks_config" {
-    for_each = var.master_authorized_networks_config
-    content {
-      dynamic "cidr_blocks" {
-        for_each = master_authorized_networks_config.value.cidr_blocks
-        content {
-          cidr_block   = lookup(cidr_blocks.value, "cidr_block", "")
-          display_name = lookup(cidr_blocks.value, "display_name", "")
-        }
-      }
-    }
-  }
-
-  dynamic "database_encryption" {
-    for_each = var.database_encryption
-
-    content {
-      key_name = database_encryption.value.keyName
-      state    = database_encryption.value.state
-    }
-  }
-
-  maintenance_policy {
-    # dynamic "recurring_window" {
-    #   for_each = local.cluster_maintenance_window_is_recurring
-    #   content {
-    #     start_time = var.maintenance_start_time
-    #     end_time   = var.maintenance_end_time
-    #     recurrence = var.maintenance_recurrence
-    #   }
-    # }
-
-    daily_maintenance_window {
-      #   for_each = try([var.maintenance_policy.window.dailyMaintenanceWindow], [])
-      #   content {
-
-      start_time = var.maintenance_policy.window.dailyMaintenanceWindow.startTime
-      #   }
-    }
-
-    # dynamic "maintenance_exclusion" {
-    #   for_each = var.maintenance_exclusions
-    #   content {
-    #     exclusion_name = maintenance_exclusion.value.name
-    #     start_time     = maintenance_exclusion.value.start_time
-    #     end_time       = maintenance_exclusion.value.end_time
-
-    #     dynamic "exclusion_options" {
-    #       for_each = maintenance_exclusion.value.exclusion_scope == null ? [] : [maintenance_exclusion.value.exclusion_scope]
-    #       content {
-    #         scope = exclusion_options.value
-    #       }
+resource "google_container_cluster" "private" {
+  for_each = { for subnet in var.subnetworks : subnet.name => subnet }
```

Comment:
Why is subnet being used to create multiple clusters? Or am I misundertanding something here?